v2/tools/generator/internal/armconversion/convert_to_arm_function_builder.go

```diff
@@ -387,6 +388,40 @@ func (builder *convertToARMBuilder) convertReferenceProperty(_ *astmodel.Convers
 	return []dst.Stmt{armIDLookup, returnIfNotNil, result}
 }
 
+// convertSecretProperty handles conversion of secret properties.
+// This function generates code that looks like this:
+//	<namehint>Secret, err := resolved.ResolvedSecrets.SecretOrErr(<source>)
+//	if err != nil {
+//		return nil, err
+//	}
+//	<destination> = <namehint>Secret
+func (builder *convertToARMBuilder) convertSecretProperty(_ *astmodel.ConversionFunctionBuilder, params astmodel.ConversionParameters) []dst.Stmt {
+	isString := astmodel.TypeEquals(params.DestinationType, astmodel.StringType)
+	if !isString {
+		return nil
+	}
+
+	isReference := astmodel.TypeEquals(params.SourceType, astmodel.SecretReferenceType)
```

Comment:
I originally read _reference_ as _pointer_, but it's actually whether it is a reference to a secret.
```suggestion
	isSecretReference := astmodel.TypeEquals(params.SourceType, astmodel.SecretReferenceType)
```

Reply:
Yeah good catch! Fixed.